src/utils/relayer.ts

```diff
@@ -0,0 +1,110 @@
+import { Redis } from 'ioredis'
+import { HttpErrors } from '@loopback/rest'
+import { Node } from '@pokt-network/pocket-js'
+
+import { BlockchainsRepository } from '../repositories'
+import { SyncCheckOptions } from '../services/sync-checker'
+import { DEFAULT_BLOCK_LOGS_LIMIT } from '../utils/constants'
+import { BlockchainDetails } from './types'
+
+// Fetch node client type if Ethereum based
+export async function fetchClientTypeLog(
+  redis: Redis,
+  blockchainID: string,
+  id: string | undefined
+): Promise<string | null> {
+  const clientTypeLog = await redis.get(`${blockchainID}-${id}-clientType`)
+
+  return clientTypeLog
+}
+
+export function isCheckPromiseResolved(promise: PromiseSettledResult<Node[]>): boolean {
+  return promise.status === 'fulfilled' && promise.value !== undefined && promise.value.length > 0
+}
+
+export function filterCheckedNodes(syncCheckNodes: Node[], chainCheckedNodes: Node[]): Node[] {
```

Comment:
you can directly return here, the variable is not neccesary as I already know from the types that the fn must return  an array of nodes